python/cuml/ensemble/randomforestclassifier.pyx

```diff
@@ -287,7 +287,9 @@ class RandomForestClassifier(Base):
         self.n_streams = handle.getNumInternalStreams()
         self.seed = seed
         if ((seed is not None) and (n_streams != 1)):
-            warnings.warn("Random seed requires n_streams=1.")
+            warnings.warn("Random seed requires n_streams=1. If n_streams"
```

Comment:
Hmm.... I don't think that's 100% accurate that we require n_streams==1 right? Maybe it should be "For reproducible results, n_streams==1 is recommended. If n_streams is > 1, results may vary due to stream/thread timing differences, even when random_seed is set". WDYT?

Reply:
We need n_streams=1 for the results to be exactly reproducible. If it is not 1, the resultant labels might be similar to when the seed is not set at all.
I like the update you suggested, it explains why n_streams > 1 will not work.

Reply:
I think the key thing in the comment here is that it would not be accurate to say `n_streams==1 is required` when a seed is set. Rather, `n_streams==1 is recommended` so you can have repeatable behavior. But you won't get a crash or anything if you set n_streams=2 and set a seed, just non-reproducible results.